File: SWP391-Vaccine-Schedule-LinhVX/SWP391/src/constants/PaymentStatus.js
export const PaymentStatus = {
    PENDING: 'PENDING',
    PROCESSING: 'PROCESSING',
    COMPLETED: 'COMPLETED',
    DECLINED: 'DECLINED',
    CANCELLED: 'CANCELLED',
    REFUNDED: 'REFUNDED',
    PARTIALLY_REFUNDED: 'PARTIALLY_REFUNDED',
    FAILED: 'FAILED',
    EXPIRED: 'EXPIRED',
    ON_HOLD: 'ON_HOLD',
    DISPUTED: 'DISPUTED'
};

export const PaymentStatusLabels = {
    [PaymentStatus.PENDING]: 'Pending',
    [PaymentStatus.PROCESSING]: 'Processing',
    [PaymentStatus.COMPLETED]: 'Completed',
    [PaymentStatus.DECLINED]: 'Declined',
    [PaymentStatus.CANCELLED]: 'Cancelled',
    [PaymentStatus.REFUNDED]: 'Refunded',
    [PaymentStatus.PARTIALLY_REFUNDED]: 'Partially Refunded',
    [PaymentStatus.FAILED]: 'Failed',
    [PaymentStatus.EXPIRED]: 'Expired',
    [PaymentStatus.ON_HOLD]: 'On Hold',
    [PaymentStatus.DISPUTED]: 'Disputed'
};

export const PaymentStatusColors = {
    [PaymentStatus.PENDING]: 'warning',
    [PaymentStatus.PROCESSING]: 'info',
    [PaymentStatus.COMPLETED]: 'success',
    [PaymentStatus.DECLINED]: 'error',
    [PaymentStatus.CANCELLED]: 'default',
    [PaymentStatus.REFUNDED]: 'secondary',
    [PaymentStatus.PARTIALLY_REFUNDED]: 'secondary',
    [PaymentStatus.FAILED]: 'error',
    [PaymentStatus.EXPIRED]: 'default',
    [PaymentStatus.ON_HOLD]: 'warning',
    [PaymentStatus.DISPUTED]: 'error'
}; 

File: Vaccination_Schedule_Children_Tracking_Project/src/main/java/com/swp_group03/vaccination/vaccination_schedule_children_tracking_project/entity/PaymentStatus.java
package com.swp_group03.vaccination.vaccination_schedule_children_tracking_project.entity;

/**
 * Enum representing the various states a payment can be in throughout its lifecycle
 */
public enum PaymentStatus {
    /**
     * Payment has been initiated but not yet processed
     */
    PENDING,
    
    /**
     * Payment is currently being processed by the payment gateway
     */
    PROCESSING,
    
    /**
     * Payment has been successfully completed
     */
    COMPLETED,
    
    /**
     * Payment was declined by the payment gateway or financial institution
     */
    DECLINED,
    
    /**
     * Payment was cancelled by the user before processing
     */
    CANCELLED,
    
    /**
     * Payment was refunded back to the user
     */
    REFUNDED,
    
    /**
     * Payment was partially refunded back to the user
     */
    PARTIALLY_REFUNDED,
    
    /**
     * Payment failed due to technical or other issues
     */
    FAILED,
    
    /**
     * Payment has expired (e.g., pending payment not completed within timeframe)
     */
    EXPIRED,
    
    /**
     * Payment is on hold pending review or additional verification
     */
    ON_HOLD,
    
    /**
     * Payment is disputed by the customer
     */
    DISPUTED
} 